### ISSUE TITLE
Binary release with Nutika

### ISSUE THREAD

**Issue by BigB84** [FelixFrog/pdfgrabber#66]
https://github.com/Nuitka/Nuitka is a compilator for python3

Implementing this may benefit in performance as python is generally slow interpreted language.

This should not be a huge problem.
Take a look at [monophony Makefile](https://gitlab.com/zehkira/monophony/-/blob/master/source/Makefile?ref_type=heads) that made it.

Although I could consider it only if project is released. See #65

**Comment by FelixFrog** [comment]
Thank for letting me know about it. I wasn't aware of its existence. Sadly I think this will add a lot of complexity for very little benefit. As of now pdfgrabber's speed is capped by the downloading speed, pymupdf's speed (which is already a c binding) and generally being single-threaded. It also consumes a lot of memory, due to often storing many 100+ mb zip files in memory. I think fixing these issues would give a much better UX improvement for the effort it would require.

Also as of now I am looking into adding a CLI interface. Once that has been added, it would be much easier to create a binary release with a simple GUI, in particular for windows (even if I am reluctant about that, I have to acknowledge that the majority of the users here are using Windows and are not particularly expert of the CLI).

**Comment by BigB84** [comment]
Right!

When it comes to GUI. I would think about it. With ease-of-use comes popularization. This app may be controversial for service owners and copyright holders. But you decide ofc.

CLI interface is good idea. Would you eventually accept PR for that?

Maybe you could open a milestone for that?

**Comment by BigB84** [comment]
Maybe there could be an separate issue (or rename that) / milestone opened for optimizations?

For unzipping for instance I've heard somewhere that WinZip is hardware accelerated, so for instance some bindings to winzip in python could be found and implemented.

Also something for simultaneous reading and writing like  [this one in go](https://github.com/djherbis/fscache)

**Comment by FelixFrog** [comment]
> CLI interface is good idea. Would you eventually accept PR for that?
Yes, ofc. For now the main interface has very low code quality, and I would like to eventually rewrite it (using sqlite as database instead of tinydb). But having a CLI interface would not stand in the way of that, and i would be a great idea regardless. Implementing it with argparser would make using something like [this](https://github.com/chriskiehl/Gooey) very trivial.

> For unzipping for instance I've heard somewhere that WinZip is hardware accelerated, so for instance some bindings to winzip in python could be found and implemented.
I will look into it. As of now my choice is towards [py7zr](https://github.com/miurahr/py7zr). The only real place where this is needed though is in the RPLUS_EPUB extraction, so I'm not too sure about adding it as a dependency just for that, maybe there is a way of using directly 7z if installed, otherwise falling back to the pythonic way.
